bin/cli.class.php

```diff
@@ -430,6 +430,14 @@ public function run() {
         continue;
       }
       $this->row++;
+      if ($this->row % 1000 == 0) {
+        // Reset PEAR_DB_DATAOBJECT cache to prevent memory leak
+        $GLOBALS['_DB_DATAOBJECT']['RESULTS'] = array();
```

Comment:
Thank you Eileen!  `CRM_Core_DAO::freeResult()` looks slower but functionally equivalent - that is, there's still no "disable the $_resultFields" option.  However, that method might result in lower memory use.  I have some massive datasets at the moment, so next week when I'm not mid-import, I will do speed and RAM usage tests to compare.